app/controllers/admin/items_controller.rb

```diff
@@ -6,22 +6,40 @@ def index
 
     def new
       @item = Item.new
+      @categories = Category.all
     end
 
     def create
-      Item.create(item_params)
-      redirect_to admin_items_path
+      item = Item.create(item_params)
+      if params[:category_list] == nil
+        flash[:errors] = "You must select a category"
+        redirect_to new_admin_item_path
+      else
+        string_category_tags = params[:category_list][:categories].uniq
+        found_tags = string_category_tags.collect{|name| Category.find_or_create_by(name: name)}
```

Comment:
Line is too long. [96/80]<br>Prefer `map` over `collect`.<br>Space missing to the left of {.<br>Space between { and | missing.<br>Space missing inside }.